test/Topologies/distributed.jl

```diff
@@ -0,0 +1,20 @@
+using MPI
+
+function runmpi(file; ntasks = 1)
+    # Some MPI runtimes complain if more resources are requested
+    # than available.
+    if MPI.MPI_LIBRARY == MPI.OpenMPI
+        oversubscribe = `--oversubscribe`
+    else
+        oversubscribe = ``
+    end
+    MPI.mpiexec() do cmd
+        Base.run(
+            `$cmd $oversubscribe -np $ntasks $(Base.julia_cmd()) --startup-file=no --project=$(Base.active_project()) $file`;
+            wait = true,
+        )
+        true
+    end
+end
+
+runmpi(joinpath(@__DIR__, "dtopo.jl"), ntasks = 4)
```

Comment:
It would probably be helpful if we could mock the ClimaComms context so that we can run the tests in the same process.

Reply:
See https://github.com/CliMA/ClimaComms.jl/issues/7.